docs/modules/ROOT/pages/usage.adoc

```diff
@@ -67,88 +67,76 @@ and from outside:
 
 where `<node-ip>` is the IP address of the node running your master.
 
-== Configuration properties
-
-There are three levels of configuration:
-
-[cols="1,1"]
-|===
-|Common properties
-|Contains configuration that is shared within the whole cluster. For example the spark version, secrets, encryption or logging options.
-
-|Role properties
-|This configuration is shared for all nodes of a certain type (Master, Worker, History-Server)
-
-|Role Group properties
-|Options provided in the role group are limited to one specific role and role group.
-|===
-
-=== Common properties
-[cols="1,1,1,1"]
-|===
-|Name
-|Type
-|Description
-|Related Spark properties
-
-|version
-|string
-|The spark version used in the format: x.y.z
-|
-
-|secret
-|string
-|A secret shared between nodes and required to submit applications via spark-submit
-|spark.authenticate=true, spark.authenticate.secret=secret;
-
-|maxPortRetries
-|integer
-|Maximum number of retries when binding to a port before giving up. When a port is given a specific value (non 0), each subsequent retry will increment the port used in the previous attempt by 1 before retrying. This essentially allows it to try a range of ports from the start port specified to port + maxRetries.
-|spark.port.maxRetries
-|===
-
-=== Role properties
-T.b.d.
-
-=== Role Group properties
-==== Master
-[cols="1,1,1,1"]
-|===
-|Name
-|Type
-|Description
-|Related Spark properties
-|===
-
-==== Worker
-[cols="1,1,1,1"]
-|===
-|Name
-|Type
-|Description
-|Related Spark properties
-
-|cores
-|integer
-|Total number of cores to allow Spark jobs to use on the machine (default: all available cores).
-|SPARK_WORKER_CORES
-
-|memory
-|string
-|Total amount of memory to allow Spark jobs to use on the machine, e.g. 1000M, 2G (default: total memory minus 1 GB).
-|SPARK_WORKER_MEMORY
-|===
-
-==== History Server
-[cols="1,1,1,1"]
-|===
-|Name
-|Type
-|Description
-|Related Spark properties
-
-|storePath
-|string
-|A local directory where to cache application data. If set, the history server will store application data on disk instead of keeping it in memory. The data written to disk will be re-used in the event of a history server restart.
-|spark.history.store.path
-|===
+== Configuration & Environment Overrides
+
+The cluster definition also supports overriding configuration properties and environment variables, either per role or per role group, where the more specific override (role group) has precedence over the less specific one (role).
+
+IMPORTANT: Overriding certain properties (such as `spark.port.maxRetries`) which are set by operator can lead to faulty installations.
```

Comment:
This is not a fitting example, i would go for `SPARK_MASTER_PORT` or `spark.master.port`

Reply:
Why is it not fitting?  The spark master port is addressed also in the environment section.

Reply:
Because that property wont lead to faulty installations ever.

Reply:
and yeah i commented before i was don at the master port... :D